api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -991,4 +995,272 @@ public void checkIfPersonAttributeTypesAreLocked() {
+	}
+	private void mergeRelationships(Person preferred, Person notPreferred, PersonMergeLogData mergedData) {
+		// copy all relationships
+		Set<String> existingRelationships = new HashSet<>();
+		// fill in the existing relationships with hashes
+		for (Relationship rel : getRelationshipsByPerson(preferred)) {
+			existingRelationships.add(relationshipHash(rel, preferred));
+		}
+		// iterate over notPreferred's relationships and only copy them if they are needed
+		for (Relationship rel : getRelationshipsByPerson(notPreferred)) {
+			if (!rel.getVoided()) {
+				boolean personAisPreferred = rel.getPersonA().equals(preferred);
+				boolean personAisNotPreferred = rel.getPersonA().equals(notPreferred);
+				boolean personBisPreferred = rel.getPersonB().equals(preferred);
+				boolean personBisNotPreferred = rel.getPersonB().equals(notPreferred);
+				String relHash = relationshipHash(rel, notPreferred);
+				
+				if ((personAisPreferred && personBisNotPreferred) || (personBisPreferred && personAisNotPreferred)) {
+					// void this relationship if it's between the preferred and notPreferred patients
+					voidRelationship(rel, "person " + (personAisNotPreferred ? "A" : "B")
+					        + " was merged to person " + (personAisPreferred ? "A" : "B"));
+				} else if (existingRelationships.contains(relHash)) {
+					// void this relationship if it already exists between preferred and the other side
+					voidRelationship(rel, "person " + (personAisNotPreferred ? "A" : "B")
+					        + " was merged and a relationship already exists");
+				} else {
+					// copy this relationship and replace notPreferred with preferred
+					Relationship tmpRel = rel.copy();
+					if (personAisNotPreferred) {
+						tmpRel.setPersonA(preferred);
+					}
+					if (personBisNotPreferred) {
+						tmpRel.setPersonB(preferred);
+					}
+					log.debug("Copying relationship " + rel.getRelationshipId() + " to " + preferred.getPersonId());
+					Relationship persisted = saveRelationship(tmpRel);
+					mergedData.addCreatedRelationship(persisted.getUuid());
+					// void the existing relationship to the notPreferred
+					voidRelationship(rel, "person " + (personAisNotPreferred ? "A" : "B")
+					        + " was merged, relationship copied to #" + tmpRel.getRelationshipId());
+					// add the relationship hash to existing relationships
+					existingRelationships.add(relHash);
+				}
+				mergedData.addVoidedRelationship(rel.getUuid());
+			}
+		}
+	}
+	
+	
+	
+	
+	private void mergeDateOfDeath(Person preferred, Person notPreferred, PersonMergeLogData mergedData) {
+		mergedData.setPriorDateOfDeath(preferred.getDeathDate());
+		if (preferred.getDeathDate() == null) {
+			preferred.setDeathDate(notPreferred.getDeathDate());
+		}
+		
+		if (preferred.getCauseOfDeath() != null) {
+			mergedData.setPriorCauseOfDeath(preferred.getCauseOfDeath().getUuid());
+		}
+		if (preferred.getCauseOfDeath() == null) {
+			preferred.setCauseOfDeath(notPreferred.getCauseOfDeath());
```

Comment:
Since you are checking preferred.getCauseOfDeath() != null in the previous if block this can be the **else**  statement. No need to check preferred.getCauseOfDeath() == null again